src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -365,7 +372,14 @@ where
                                     }
                                     untimestamped_messages.entry(pid).or_default().push((message, offset));
                                 }
-                                SourceMessageType::SourceStatus(_update) => {
+                                SourceMessageType::SourceStatus(update) => {
+                                    let update = match update {
+                                        SourceStatusUpdate { status: SourceStatus::Starting, ..} => Some(HealthStatus::Starting),
+                                        SourceStatusUpdate { status: SourceStatus::Running, ..} => Some(HealthStatus::Running),
+                                        SourceStatusUpdate { status: SourceStatus::Stalled, error: Some(e)} => Some(HealthStatus::StalledWithError(e)),
+                                        _ => None,
```

Comment:
Thoughts on expanding the `_` for future compat?

also I think this `match` adds some good support for merging `error` into the `status`: what is the right behavior for `SourceStatusUpdate { status: SourceStatus::Stalled, error: None }`?  Should it be `StalledWithError("")` instead of just dropping it?

Reply:
Since "this should never happen" on the current branch, I've added a proper catchall with a warn log to start with.